src/scraper_net.cpp

```diff
@@ -531,6 +542,15 @@ bool CScraperManifest::addManifest(std::unique_ptr<CScraperManifest>&& m, CKey&
 
     // Call manifest complete to notify peers of new manifest.
     manifest.Complete();
+
+    // lock cs_ConvergedScraperStatsCache and mark ConvergedScraperStatsCache dirty because a new manifest is present,
+    // so the convergence may change.
+    {
+        LOCK(cs_ConvergedScraperStatsCache);
+
+        ConvergedScraperStatsCache.bClean = false;
```

Comment:
Do we also need to invalidate the cache for `CScraperManifest::DeleteManifest()` (if a node purges it's unauthorized manifests)?

Reply:
You know, I think you are right, and the best place to do it is in the DeleteManifest() function.